src/common/marlin_server.cpp

```diff
@@ -76,6 +76,7 @@ typedef struct _marlin_server_t {
     uint32_t command_end;                                    // variable for notification
     marlin_mesh_t mesh;                                      // meshbed leveling
     uint64_t mesh_point_notsent[MARLIN_MAX_CLIENTS];         // mesh point mask (points that are not sent)
+    uint8_t device_state_change;                             // marks that change in device's printing state has occured
```

Comment:
server should have a variable which indicates the state, like gcode_name[..]
when the state changes, you have to change this variable (with values from DEVICE_STATES_t)
and there can be a variable in client vars with state. and just like other variables change here the value on change eg. z_axix

Reply:
Its a bit more complicated. marlin_server.vars has device_state that contains DEVICE_STATES_t values. This variable is just a helper that indicates a change (some of the changes are user's action so update_vars function has no way of knowing if device_state variable should be updated)